roles/docker/templates/docker.service.j2

```diff
@@ -18,7 +18,7 @@ Environment=GOTRACEBACK=crash
 ExecReload=/bin/kill -s HUP $MAINPID
 Delegate=yes
 KillMode=process
-ExecStart={{ docker_bin_dir }}/docker daemon \
+ExecStart={{ docker_bin_dir }}/docker{% if docker_version.stdout|version_compare('17.03', '<') %} daemon{% else %}d{% endif %} \
```

Comment:
I would prefer to get rid of complex logic in templates whenever possible. Can you do the following instead?
Modify roles/docker/defaults/main.yml and add the following line:
```
docker_daemon_binary: >-
  {%- if docker_version.stdout|version_compare('17.03', '<') -%}
  docker daemon
  {%- else -%}
  dockerd
  {%- endif -%}
```
And then this line should read in the systemd unit file:
ExecStart={{ docker_bin_dir }}/{{ docker_daemon_binary }}

Reply:
Hold on... we can't rely on docker_version.stdout in a defaults file. Let's just use your version as-is for now.